static/js/browser/kg.js

```diff
@@ -938,63 +938,60 @@ function renderKGPage(
 
 /** Page setup after initial loading */
 window.onload = () => {
-  // Render a set of data
-  const provs = util.sendRequest(`/node/triples?dcids=Provenance`, false)[
-    "Provenance"
-  ];
-  let provDomain = {};
-  for (let prov of provs) {
-    if (prov["predicate"] == "typeOf" && !!prov["subjectName"]) {
-      provDomain[prov["subjectId"]] = new URL(prov["subjectName"]).host;
-    }
-  }
-
   const dcid = document.getElementById("bg-dcid").textContent;
+  const provPromise = getTriples("Provenance");
+  const triplesPromise = getTriples(dcid);
+
+  Promise.all([provPromise, triplesPromise]).then(([provs, triples]) => {
```

Comment:
since can not may window.onload async, here did a bit more change to use Promise.all instead of two sync block call. Performance should be improved.

Reply:
I suppose we're trading 2 sequential calls direct to the mixer with parallel calls to flask (which is now having to decode and re-encode large json blobs which was my concern).

Reply:
Yes, these are raw response which are large. Ideally the data should be processed in the server as we do for place page.